scripts/train_agents.py

```diff
@@ -2,16 +2,200 @@
+        args.learner_type = adversary_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = root_adv + '/' + adversary_type + '/' + str(round)
+        ADV(args=args,
+            agent_folder_path = root_adv + '/' + main_agent_type + '-' + adversary_type + 'play' + + str(round-1), 
+            agent_file_tag = pwadv_tag + '/' + CheckedPoints.FINAL_TRAINED_MODEL)
+        
+        how_long = how_long_init + how_long_for_agent*(round+1)
+        set_input(args=args, quick_test=quick_test, how_long=how_long)
+        args.learner_type = main_agent_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = root_adv + '/' + main_agent_type + '-' + adversary_type + 'play' + '/' + str(round)
+        PwADVs( args=args, 
+                agent_folder_path = root_adv + '/' + main_agent_type + '-' + adversary_type + 'play' + '/' + str(round-1), 
+                agent_file_tag = pwadv_tag + '/' + CheckedPoints.FINAL_TRAINED_MODEL,
+                adv_folder_paths = [root_adv + '/' + adversary_type + '/' + str(round)], 
+                adv_file_tag = adv_tag + '/' + checked_adversary)
+        
+def MultiAdversaryPlay(args, 
+                        exp_tag = 'M2FP', 
+                        main_agent_path = None,
+                        main_agent_type = LearnerType.SUPPORTER, 
+                        adversary_type = LearnerType.SELFISHER, 
+                        checked_adversary = CheckedPoints.FINAL_TRAINED_MODEL, 
+                        how_long_init = 4.0,
+                        how_long_for_agent = 4.0,
+                        how_long_for_adv = 4.0,
+                        rounds_of_advplay = 401,
+                        reward_magnifier = 3.0):
+    
+    args.dynamic_reward = False
+    args.final_sparse_r_ratio = 0.5
+    if main_agent_path is None:
+        how_long = how_long_init
+        set_input(args=args, quick_test=quick_test, how_long=how_long)
+        args.SP_seed = args.ADV_seed
+        args.SP_h_dim = args.ADV_h_dim
+        args.learner_type = main_agent_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = exp_tag 
+        SP(args=args, pop_force_training=True)
+        sp_tag = 'sp_s' + args.SP_seed + '_h' + args.SP_h_dim + '_tr(SP)_ran'
+        main_agent_path = exp_tag + '/' + sp_tag
+    root = 'agent_models/' + main_agent_path
+    root_adv = root + '/' + exp_tag
+
+    adv_tag = 'adv_s' + args.ADV_seed + '_h' + args.ADV_h_dim + '_tr(H)_ran'
+    pwadv_tag = 'pwadv_s' + args.ADV_seed + '_h' + args.ADV_h_dim + '_tr(SP_SPADV)_ran'
+
+    how_long = how_long_for_adv
+    set_input(args=args, quick_test=quick_test, how_long=how_long)
+    args.learner_type = adversary_type
+    args.reward_magnifier = reward_magnifier
+    args.exp_dir = root_adv + '/' + adversary_type + '/0'
+    ADV(args=args, 
+        agent_folder_path = root, 
+        agent_file_tag= CheckedPoints.BEST_EVAL_REWARD)
+
+    how_long = how_long_init + how_long_for_agent
```

Comment:
Agree to place learner_type and reward_magnifier, but still allows our user to control them. However, I prefer to have how_long outside there because it is clearer to see how long we take to train each of them.

In addition, I need to confirm whether we can pass exp_dir directly into the function. I will get back to you on this one.